core/arch/arm/plat-imx/imx6ul.c

```diff
@@ -32,4 +33,5 @@ static void init_csu(void)
 void plat_cpu_reset_late(void)
 {
 	init_csu();
+	imx_configure_tzasc();
```

Comment:
Do you think it would be possible to move imx_configure_tzasc out of the plat_cpu_reset_late
and add driver_init(imx_configure_tzasc); in the tzx380.c file?
this way only compile tzc support modification is required in order to configure the regions?

Reply:
please note that this method have currently a drawback.
tzc will be configured after the memoy mapping of the devicetree tree. This mapping is done in NS and will result in an error since be defaut the region 0 of the tzasc380 is configured for S access only.
in my case this is not an issue since the tzc region 0 is configure for ns access as well before entering OP-TEE.

Reply:
Yes, this is the exact reason why the autoconfiguration is done in `plat_cpu_reset_late`.
U-Boot may have support for the TZC380 controller, but I am reluctant to add support for barebox, since the TZC should really be configured by the secure world operating system.
If an early load of OP-TEE is done during the barebox pre-bootloader entry into main barebox, configuration can probably only be done in the DCD table.
I'd really like to keep the configuration and requirements at one place.

Reply:
you probably are already doing  things for the  TZASC (enabling the controller when not accessing DDR) through DCD table (or plugin). Adding the configuration for the Region 0 should probaly be the same.
it could also be good to transform the plat_cpu_reset_late into som kind of init_call function but not sure it is worth it.

Reply:
@Emantor Just wonder, pre-bootloader for barebox is something like U-Boot SPL?

Reply:
Yes.